pytorch/training/docker/1.6.0/py3/cu101/Dockerfile.gpu

```diff
@@ -71,17 +76,45 @@ RUN wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86
  && apt-get install -f -y \
  && rm libcublas10_${CUBLAS_VERSION}.deb
 
-RUN wget https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-$OPEN_MPI_VERSION.tar.gz \
- && gunzip -c openmpi-$OPEN_MPI_VERSION.tar.gz | tar xf - \
- && cd openmpi-$OPEN_MPI_VERSION \
- && ./configure --prefix=/home/.openmpi \
- && make all install \
- && cd .. \
- && rm openmpi-$OPEN_MPI_VERSION.tar.gz \
- && rm -rf openmpi-$OPEN_MPI_VERSION
-
-ENV PATH="$PATH:/home/.openmpi/bin"
-ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/home/.openmpi/lib/"
+# RUN wget https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-$OPEN_MPI_VERSION.tar.gz \
+#  && gunzip -c openmpi-$OPEN_MPI_VERSION.tar.gz | tar xf - \
+#  && cd openmpi-$OPEN_MPI_VERSION \
+#  && ./configure --prefix=/home/.openmpi \
+#  && make all install \
+#  && cd .. \
+#  && rm openmpi-$OPEN_MPI_VERSION.tar.gz \
+#  && rm -rf openmpi-$OPEN_MPI_VERSION
+
+# ENV PATH="$PATH:/home/.openmpi/bin"
+# ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/home/.openmpi/lib/"
+
+# Install OpenMPI & EFA
+RUN cd /tmp && \
+  curl -O https://s3-us-west-2.amazonaws.com/aws-efa-installer/aws-efa-installer-1.8.4.tar.gz && \
+  tar -xf aws-efa-installer-1.8.4.tar.gz && \
+  cd aws-efa-installer && \
+  ./efa_installer.sh -y --skip-kmod --skip-limit-conf --no-verify && \
+  cd /tmp && \
+  rm -rf aws-efa-installer && \
+  rm -rf aws-efa-installer-1.8.4.tar.gz && \
+  echo export PATH="$OPEN_MPI_PATH/bin:$PATH" >> /root/.bashrc && \
+  echo export LD_LIBRARY_PATH="$OPEN_MPI_PATH/lib:$AMAZON_EFA_PATH/lib:$LD_LIBRARY_PATH" >> /root/.bashrc
+
+RUN ["/bin/bash", "-c", "source /root/.bashrc"]
```

Comment:
Is it necessary to run this source command with list-format arguments?

Reply:
source command fails outside bash. Docker used some other shell so I had to enclose this in list-format to make it work with bash shell